generator/generator.go

```diff
@@ -408,21 +409,53 @@ func (g *generator) Build(
 		}
 	}
 
-	if !hasPointerStructMethod && ((source.Named && !source.Basic) || (target.Named && !target.Basic) || (source.Pointer && source.PointerInner.Named && !source.PointerInner.Basic)) {
+	isSourceNamed, isTargetNamed := source.Named && !source.Basic, target.Named && !target.Basic
+	isSourcePointerNamed := source.Pointer && source.PointerInner.Named && !source.PointerInner.Basic
+	isTargetPointerNamed := target.Pointer && target.PointerInner.Named && !target.PointerInner.Basic
+	if !hasPointerStructMethod && (isSourceNamed || isTargetNamed || isSourcePointerNamed || isTargetPointerNamed) {
 		name := g.namer.Name(source.UnescapedID() + "To" + strings.Title(target.UnescapedID()))
 
+		// Prepare method fields using context fields as a base.
+		methodFields := map[string]*builder.FieldMapping{}
+		for fieldName, fieldValue := range ctx.Fields {
+			// Copy all context fields but skip ignored fields.
+			// This adds all fields to the possibly reused method.
+			// See `struct_pointer_nested_ignore.yml` test case.
+			if !fieldValue.Ignore {
+				methodFields[fieldName] = fieldValue
+			}
+		}
+
+		// Prepare method flags using context flags as a base.
+		// Use only certain flag types from the context flags.
+		methodFlags := builder.ConversionFlags{}
+		for flagType, flagStatus := range ctx.Flags {
+			switch flagType {
+			case builder.FlagWrapErrors:
+				methodFlags[flagType] = flagStatus
+			case builder.FlagMatchIgnoreCase:
+				methodFlags[flagType] = flagStatus
+			case builder.FlagIgnoreMissing:
+				// See `ignore_missing_no_propagate.yml` test case.
+			case builder.FlagIgnoreUnexported:
+				// See `ignore_missing_no_propagate.yml` test case.
+			}
+		}
+		for flagType, flagStatus := range g.flags {
+			methodFlags[flagType] = flagStatus
+		}
+
 		method := &methodDefinition{
 			ID:     name,
 			Name:   name,
 			Source: xtype.TypeOf(source.T),
 			Target: xtype.TypeOf(target.T),
-			Fields: map[string]*builder.FieldMapping{},
-			Flags:  g.flags,
+			Fields: methodFields,
```

Comment:
Please pay attention on this. I'm worrying about side effects, because this method is picked up from the cache by signature, but signature is unaware of method flags and fields.

Without this, `govertor` fails on copying renamed or different case fields, because internal function is not aware of renames and case match.

With this change, `govertor` will handle fields renames in internal functions, but rules of these conversions will be taken from first method who spawned this function.

Only one solution I know is to add flags and fields to the method signature, but it is a huge change, I want to ask author's opinion first.

@jmattheis

Reply:
Or, as another way, just generate this function every time without adding it to cache and lookup for in in different methods. This will not add a lot of code to the result binary, but will significantly reduce generator complexity.

Reply:
Both the method flags and method fields should _not_ be propagated to other generator methods, as this adds possible clashes with other methods. Example:
```
// goverter:converter
type Converter interface {
    // goverter:matchIgnoreCase
	Convert1(Input1) Output
	Convert2(Input2) Output
}

type Input1 struct { Inner InputInner }
type Input2 struct { Inner InputInner }
type Output struct { Inner OutputInner }
type InputInner struct { Age int }
type OutputInner struct { AGE int }
```
Now it's unclear if the internal generated method converting from InputInner to OutputInner should have matchIgnoreCase enabled or not, therefore the flags should not propagate, and the user should add the flag like this:
```
// goverter:converter
type Converter interface {
	Convert1(Input1) Output
	Convert2(Input2) Output

	// goverter:matchIgnoreCase
	ConvertInner(InputInner) OutputInner
}
```
this is also described here: https://goverter.jmattheis.de/#/conversion/configure-nested

Reply:
Both the method flags and method fields should not be propagated to other generator methods, as this adds possible clashes with other methods. Example:
```go
// goverter:converter
type Converter interface {
    // goverter:matchIgnoreCase
	Convert1(Input1) Output
	Convert2(Input2) Output
}

type Input1 struct { Inner InputInner }
type Input2 struct { Inner InputInner }
type Output struct { Inner OutputInner }
type InputInner struct { Age int }
type OutputInner struct { AGE int }
```
Now it's unclear if the internal generated method converting from InputInner to OutputInner should have matchIgnoreCase enabled or not, therefore the flags should not propagate, and the user should add the flag like this:
```go
// goverter:converter
type Converter interface {
	Convert1(Input1) Output
	Convert2(Input2) Output

	// goverter:matchIgnoreCase
	ConvertInner(InputInner) OutputInner
}
```
this is also described here: https://goverter.jmattheis.de/#/conversion/configure-nested

Reply:
Thank you for the clarification. That means I was wrong with this.

Reply:
Got rid of this.